lib/modules/manager/pub/extract.spec.ts

```diff
@@ -103,6 +123,48 @@ describe('modules/manager/pub/extract', () => {
             datasource: dartDatasource,
             skipReason: 'path-dependency',
           },
+          {
+            currentValue: '',
+            depName: 'git_package',
+            packageName: 'https://github.com/some-url/some-package',
+            depType: dependenciesDepType,
+            datasource: gitRefDatasource,
+            skipReason: 'unspecified-version',
+          },
+          {
+            currentValue: 'v1.0.0',
+            depName: 'git_package_ref',
+            packageName: 'https://github.com/some-url/some-package-ref',
+            depType: dependenciesDepType,
+            datasource: gitRefDatasource,
+            skipReason,
+          },
+          {
+            currentValue: '^1.1.0',
+            depName: 'git_package_version',
+            packageName: 'https://github.com/some-url/some-package-version',
+            depType: dependenciesDepType,
+            datasource: gitRefDatasource,
+            skipReason,
+          },
+          {
+            currentValue: '^1.1.0',
+            depName: 'git_package_version_extract',
+            packageName:
+              'https://github.com/some-url/some-package-version-extract',
+            depType: dependenciesDepType,
+            datasource: gitRefDatasource,
+            extractVersion: 'v',
```

Comment:
Have you tested this?  

AFAIK this shouldn't work, but rather should be `^v.*`

Reply:
I'm refering to if you have tested this specific combination because [extractVersion doesn't work like that](
https://docs.renovatebot.com/configuration-options/#extractversion) and `semver-coerced` should handle this automatically. 

```suggestion
```